bin/einhorn

```diff
@@ -190,14 +208,19 @@ if true # $0 == __FILE__
 
   optparse = OptionParser.new do |opts|
     opts.on('-b ADDR', '--bind ADDR', 'Bind an address and add the corresponding FD via the environment') do |addr|
-      unless addr =~ /\A([^:]+):(\d+)((?:,\w+)*)\Z/
-        raise "Invalid value for #{addr.inspect}: bind address must be of the form address:port[,flags...]"
+      unless addr =~ BIND_PATTERN
+        raise "Invalid value for #{addr.inspect}: bind address must be of the form address:port[,flags...] or /path/to/unix/socket[,flags...]"
+      end
+
+      flags = $~["flags"].split(",").reject(&:empty?).map(&:downcase)
```

Comment:
We can bind the match to a variable, instead of `$~` but this was the original patch